framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -135,16 +140,20 @@ public static ContractsUtils getInstance(GenericAnnotatedTypeFactory<?, ?, ?, ?>
         if (enforcedQualifier == null) {
             return Collections.emptySet();
         }
-        Set<Contract> result = new LinkedHashSet<>();
+        Set<T> result = new LinkedHashSet<>();
         List<String> expressions =
                 AnnotationUtils.getElementValueArray(
                         contractAnnotation, "expression", String.class, false);
         Boolean annoResult =
                 AnnotationUtils.getElementValueOrNull(
                         contractAnnotation, "result", Boolean.class, false);
         for (String expr : expressions) {
-            result.add(
-                    Contract.create(kind, expr, enforcedQualifier, contractAnnotation, annoResult));
+            @SuppressWarnings("unchecked") // clazz consistent with enum
```

Comment:
Instead of suppressing this warning and the ErrorProne warning, you could use `clazz.cast(...)` instead.  Any reason not to do that?

Reply:
No reason, `clazz.cast` is indeed much nicer.